lldb/source/Target/StackFrameRecognizer.cpp

```diff
@@ -145,6 +167,17 @@ StackFrameRecognizerManager::GetRecognizerForFrame(StackFrameSP frame) {
       if (!entry.module_regexp->Execute(module_name.GetStringRef()))
         continue;
 
+    ConstString function_name = [&]() {
+      switch (entry.mangling_preference) {
+      case Mangled::ePreferMangled:
+        return function_name_mangled;
+      case Mangled::ePreferDemangled:
+        return function_name_demangled;
+      case Mangled::ePreferDemangledWithoutArguments:
+        return function_name_noargs;
+      }
+    }();
```

Comment:
Can't we just pass `symctx.GetFunctionName(entry.mangling_preference)` here right?

No need for the switch statement here and above and also no need to keep a map of `m_used_manglings`

Reply:
I wasn't sure about performance here. The code previously computed the demanded name only once and then checked against all frame recognizers. I am not sure how expensive it is to demangle a name

Reply:
The demangling gets chached anyway. So wouldn't worry about it

Reply:
By the way: I don't think I will be merging this part of my commit in the end. https://github.com/llvm/llvm-project/pull/105756 is very similar, and I think I will just rebase on it, after it got merged

Reply:
@vogelsgesang I closed my PR, feel free to copy&paste what you find useful.

Reply:
> The demangling gets chached anyway (in the Mangled class). So wouldn't worry about it

ah, great! I didn't know this. I removed the additional caching from `StackFrameRecognizer` again